solver/src/main/java/org/chocosolver/solver/search/strategy/selectors/variables/AbstractCriterionBasedStrategy.java

```diff
@@ -105,4 +106,22 @@ public AbsctractCriterionBasedStrategy(IntVar[] vars, long seed,
     }
 
     protected abstract double weight(IntVar v);
+
+    protected final int futVars(Propagator prop) {
+        int pid = prop.getId();
+        int futVars = pid2arity.get(pid);
+        if (futVars == -1) {
+            futVars = 0;
+            for (int i = 0; i < prop.getNbVars(); i++) {
+                if (!prop.getVar(i).isInstantiated()) {
+                    if (++futVars > 1) {
+                        break;
+                    }
+                }
+            }
```

Comment:
What is the purpose of this for loop ? As I understand it, it increases futVars until it is strictly greater than 1 and then call arity() that is basically this for loop, without the stopping criterion.

Reply:
Arf ! You're right, maybe I should stop working with my children around me

Reply:
But I claim that considering big arity constraint it is worth computing and storing that value once.

Reply:
Well, I would if this value was not moving all the time. The number of uninstantiated variables inside a propagator might be different at each node depending on the filterings that are applied.

Reply:
But `pid2arity` is cleared l.66, before iterating over variables. 
The only purpose of `pid2arity` is simply to avoid  to compute the "arity" of a propagator everytime a variable of his scope needs to know its weight.

Reply:
Yes, my bad, it is a good idea to have this map !